src/main/java/fact/utils/SamplePedestalEvent.java

```diff
@@ -0,0 +1,350 @@
+        int sampleSize = this.bins_index.get(binNum).size();
+        if (sampleSize==0) {
+            log.error("Samplesize of bin: "+binNum+" is too small");
+            return null;
+        }
+        int rand = this.randGenerator.nextInt(sampleSize);
+        int index = this.bins_index.get(binNum).get(rand);
+        Data item = null;
+        try {
+            item = this.getNoiseEvent(index);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        // insert the result into the input item
+        for (String s : item.keySet()) {
+            input.put(this.prependKey+s, item.get(s));
+        }
+
+        return input;
+    }
+
+
+    public void setNoiseCondition(String noiseCondition) {
+        this.noiseCondition = noiseCondition;
+        ConditionFactory factory = new ConditionFactory();
+        this.condition = factory.create(noiseCondition);
+    }
+
+
+    public void setBinning(String[] binning) {
+        this.binning = binning;
+        if (binning.length==1) { // single value
+            double steplength = Double.parseDouble(this.binning[0]);
+            int numBins = (int)ceil(90.0/steplength);
+            this.bins = new double[numBins];
+            for (int i=0; i<numBins; i++) {
+                this.bins[i] = steplength*(i+1)-0.5;
+            }
+        } else { // each number describs the right value of the bin, 0 not included
+            this.bins = new double[this.binning.length];
+            for (int i=0; i<this.binning.length; i++) {
+                this.bins[i] = Double.parseDouble(this.binning[i]);
+            }
+        }
+    }
+
+
+    public void setSeed(long seed) {
+        this.seed = seed;
+    }
+
+    /**
+     * Creates the fact path from the night and runID excluding the extension
+     * @param night
+     * @param runID
+     * @return the constructed path
+     */
+    private String createPathFromNightAndRunId(int night, int runID) {
+        String nightStr = String.format("%d", night);
+        String year = nightStr.substring(0,4);
+        String month = nightStr.substring(4,6);
+        String day = nightStr.substring(6,8);
+        return String.format("%s/%s/%s/%d_%03d", year, month, day, night, runID);
+    }
+
+    /**
+     * Constructs the full filename of a fact path and test if it exists.
+     * Makes also sure the extension is correct
+     * @param fullpath
+     * @return
+     */
+    private String getPathWithExt(String fullpath) {
+        // test .fz
+        File noiseFile = new File(fullpath+".fits.fz");
+        if (noiseFile.exists()) {
+            return fullpath+".fits.fz";
+        }
+        // test .gz
+        noiseFile = new File(fullpath+".fits.gz");
+        if (noiseFile.exists()) {
+            return fullpath+".fits.gz";
+        }
+        throw new RuntimeException("Couldn't find data file: "+fullpath+".fits.*");
+    }
+
+    /**
+     * Creates the path to the drs file and checks whether it exists
```

Comment:
This need explaination how the DRS file ist selected. What does it look for, the temporal closest, the last file?

Reply:
In that case: Please add an explanation what `drs0` and `drs1` are, and by which priority they are chosen for the drs calibration of the event